### PR TITLE
feat: [CVS-8890]: adding the support of deleting healthsource for verify step

### PR DESCRIPTION
**Pull request by harshilbits**
##### Summary:
1. Adding the functionality to delete Health source from the Verify step.
2. Code cleanup and refactoring.

##### Jira Links:
https://harness.atlassian.net/browse/CVS-8890

##### Screenshots:

Monitored Service ®
New Note

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by harshilbits** [comment]
retrigger jest

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/HealthSources/HealthSources.tsx [hunk 1]
```diff
@@ -40,18 +40,30 @@ export default function HealthSources(props: HealthSourcesProps): React.ReactEle
...
-            editHealthSource(tableProps?.row?.original)
-          }}
-          className={css.link}
-        />
+        <Container>
+          <Icon
+            name="edit"
+            size={14}
+            onClick={() => {
+              editHealthSource(tableProps?.row?.original)
```
Comment (praneshg239) [inline src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/HealthSources/HealthSources.tsx:52]: We can use `useCallback`, which prevents re-render of small components. Also overusing it causes unnecessary memory consumption. We shall discuss on this later.

**Review by praneshg239 — APPROVED** [review]

File: src/modules/85-cv/pages/health-source/HealthSourceTable/VerifyStepHealthSourceTable.tsx [hunk 2]
```diff
@@ -114,15 +168,23 @@ export default function VerifyStepHealthSourceTable(tableProps: VerifyStepHealth
...
+        onDeleteHealthSourceVerifyStep={selectedRow => {
+          setRowToDelete(selectedRow)
+          openDialog()
+        }}
         isRunTimeInput={isRunTimeInput}
         onAddNewHealthSource={onAddNewHealthSource}
         value={healthSourcesList as HealthSource[]}
         shouldRenderAtVerifyStep
-        onSuccess={data => {
+        onSuccess={(data: any) => {
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/HealthSourceTable/VerifyStepHealthSourceTable.tsx:187]: remove

File: src/modules/85-cv/pages/health-source/HealthSourceTable/__tests__/VerifyStepHealthSourceTable.test.tsx [hunk 3]
```diff
@@ -33,21 +39,82 @@ describe('Verify VerifyStepHealthSourceTable', () => {
...
+    const updatedProps = { ...props, isRunTimeInput: true }
     const { container } = render(
       <TestWrapper>
-        <VerifyStepHealthSourceTable {...props} />
+        <VerifyStepHealthSourceTable {...updatedProps} />
       </TestWrapper>
     )
     expect(container).toMatchSnapshot()
   })
+
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/HealthSourceTable/__tests__/VerifyStepHealthSourceTable.test.tsx:50]: move data to mock

**Review by deepesh-ui — APPROVED** [review]

**Pull request merged**